src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs

```diff
@@ -175,6 +175,206 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int IndexOfAny(ref char searchSpace, char value0, char value1, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            fixed (char* pChars = &searchSpace)
+            {
+                char* pCh = pChars;
+                char* pEndCh = pCh + length;
+
+#if !netstandard11
+                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
+                {
+                    // Figure out how many characters to read sequentially until we are vector aligned
```

Comment:
Why start with a sequential read?

You should be able to do a single unaligned read, mask out bits that would be covered by the first aligned read, and then do the vectorized check (and I'm not even sure you need to mask for this algorithm).

Reply:
I've just copied the `IndexOf<char>(this Span<char> span, char value0)` method and used the same multi-comparison with vectorized `BitwiseOr` as `byte` to widen it to 2 and 3 chars.

So an improved approach to vectorization, whether an unaligned start or using instrinsics directly should also apply to those.

Might be worth a tracking issue?

Reply:
> Might be worth a tracking issue?

Probably. I'll log one sometime Tuesday, if I don't get to it before then.

Reply:
Not sure it would help. The vectorization is fast at discounting characters; but not fast at then finding a match where the sequential by itself would probably be faster - rather than doing an overlapping vector at start and overlapping vector at the end.